### ISSUE TITLE
Document Viewer | Accessibility: use heading tags in righthand menu

### ISSUE THREAD

**Issue by abbyraskinUSDS** [department-of-veterans-affairs/caseflow#1853]
## Description
Currently our headers in the righthand menu (issue tags/comments) don't have heading tags, so there's no explicit hierarchy on the page for users with a screen reader.

## AC
- [x] Verify that the headers are tagged according to this order: 
  - H1 - Document type name (In document viewer top toolbar)
  - H2 - Annotation menu (not shown but have screen reader say it)
  - H3 - Document information
  - H3 - Categories
  - H3 - Issue tags
  - H3 - Comments
- [ ] Verify that all header tags can be heard through a screen reader

## Mock up
![image](https://cloud.githubusercontent.com/assets/13420618/26254068/5215ec26-3c83-11e7-8eb2-9bda7ec2234c.png)

H1 - Document type name (In document viewer top toolbar)
H2 - Annotation menu (not shown but have screen reader say it)
H3 - Document information
H3 - Categories
H3 - Issue tags
H3 - Comments

## Note
The order of how things should be heard through a screen reader:

- Caseflow Reader (in navigation)
- Kavi Harshawat (log in dropdown in navigation)
- Back to all documents
- Categories tab(s)
- Document type `h1`
- Zoom in/fit to screen/out/ download
- Page number
- Document number
- Annotation menu `h2`
- Document information `h3`
-  Categories tags `h3`
- Issue tags `h3`
- Comments `h3`

## Related
#2161

**Comment by gnakm** [comment]
@abbyraskin Would like your design input on this!

Right now the only headers we have for is related issues and comments but I wouldn't want a user who is using a screen reader to assuming content starts from those sections. I think issues and comments should be `h2`, which leaves us to make a `H1`. I think something like "Document Menu" is good in that it describes where you have landed. 

**_Option 1_**
Add Document Menu at top with slightly bigger font. Problem is mainly that we are running out of space!

![image](https://cloud.githubusercontent.com/assets/13420618/25880164/bcab83e8-3503-11e7-9d85-e3f49893a283.png)

**_Option 2_** 

I prefer this, since the screen reader will read in the order of "Document Menu" --> "Hide Menu"
Only concern I have is the placement of the hide menu link since we went back and forth about that placement many times before. 

![image](https://cloud.githubusercontent.com/assets/13420618/25880192/e9c46340-3503-11e7-8684-927ef9d0686b.png)

**Comment by abbyraskin** [comment]
@gnakm sorry for the slow response on this one!

Good thinking!! I like option 2 because of the vertical space constraints you mentioned. I also think it's logical for the hide menu to be in the top right, since it's a common placement for "close" actions. 

- My only concern here is that the menu label looks almost identical to the clickable button. Wondering if we could distinguish these more (more for the benefit of non screen reader users, a larger typeface could be sufficient?). It also looks like our current Hide menu and Close menu buttons are inconsistent from one another.

- The > doesn't seem very intuitive to me when it's in the top right. What do you think of using a more traditional close menu icon (like a circle with an x)? I'm definitely open to pushback on this one as it may be more of a weird user preference thing :)

-- Also, I noticed we say "Open" and "Hide." Would it make more sense to stick with either open/close or show/hide?

**Comment by abbyraskin** [comment]
sorry, more thoughts:

- Annotation Menu rather than Document Menu?
- Do we need a header for the document itself...? I wonder if that section could be confusing. The document type would work, except it's also a link. hm.

**Comment by gnakm** [comment]
@abbyraskin Eureka! I learned something new. I talked to Shane and he said that we can have a screen reader that calls out something without existing on the page. Ex. For the mockup above, before it goes through all the toolbar (<back to all docs, zoom, etc..), we can have a screen reader just read "C&P exam document" or "Document of C&P exam." Then we can make the other side menu things `h2`

I do also think we should do hide/show. Close might imply closing out the menu. I can make another issue for that including same font size.

**Comment by gnakm** [comment]
I wonder if we should also add in the screen reader "Annotation menu."
It will be helpful differentiate it from the toolbar.

**Comment by abbyraskin** [comment]
I concur!

**Comment by gnakm** [comment]
Back to `h1` what are your thoughts on the following:
- "Viewing [name of document type]" (This is the most accurate but i find this extremely ironic).
- "Reviewing [name of document type]"
- "Inside [name of doc type]"

**Comment by abbyraskin** [comment]
@gnakm Let's hold off on this until the annotation menu redesign is ready (a couple sprints out)

**Comment by astewarttistatech** [comment]
@tejans24 All of the H1, H2, and H3 tags are there but are not being read by a screenreader.

**Comment by tejans24** [comment]
@abbyraskinUSDS What's supposed to happen with these headers in correlation with the screen readers?

**Comment by abbyraskinUSDS** [comment]
The screenreader doesn't need to read the tags themselves (e.g., "H1"), but it should read through each item in heading order as listed in the AC.

**Comment by mdbenjam** [comment]
I'm moving this back to in-validation. It seems like having it read out H1 is not intended behavior.

**Comment by astewarttistatech** [comment]
H2 is currently not being read in preprod. JAWS isn't picking this up either.

**Comment by mdbenjam** [comment]
Just verified this using voice over. It seems to work for me. @astewarttistatech  can we sync on this when you return?

**Comment by astewarttistatech** [comment]
@mdbenjam I cannot seem to access this issue due to #4570 happening in UAT and Preprod being down. We can definitely sync on this when you are free.
